packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -719,6 +718,15 @@ class ScrollableState extends State<Scrollable> with TickerProviderStateMixin, R
     }
   }
 
+  bool _handleScrollMetricsNotification(ScrollMetricsNotification notification) {
+    if (notification.context == notificationContext) {
```

Comment:
Maybe we should check the depth of the events.

Reply:
can you elaborate?

Reply:
ah I see what you meant

Reply:
Oh,you already checked the context, I think it have the same effect.

Reply:
I think use depth is safer since another scrolling context can target the same notification context, that is a weird case if that happens though